libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -573,20 +575,29 @@ void NavEKF3_core::send_status_report(mavlink_channel_t chan) const
     Vector2f offset;
     getVariances(velVar, posVar, hgtVar, magVar, tasVar, offset);
 
+
     // Only report range finder normalised innovation levels if the EKF needs the data for primary
     // height estimation or optical flow operation. This prevents false alarms at the GCS if a
     // range finder is fitted for other applications
-    float temp;
+    float temp = 0;
     if (((frontend->_useRngSwHgt > 0) && activeHgtSource == AP_NavEKF_Source::SourceZ::RANGEFINDER) || (PV_AidingMode == AID_RELATIVE && flowDataValid)) {
         temp = sqrtF(auxRngTestRatio);
-    } else {
-        temp = 0.0f;
     }
-    const float mag_max = fmaxF(fmaxF(magVar.x,magVar.y),magVar.z);
+
+    const mavlink_ekf_status_report_t packet{
+        velVar,
+        posVar,
+        hgtVar,
+        fmaxF(fmaxF(magVar.x,magVar.y),magVar.z),
+        temp,
+        flags,
+        tasVar
+    };
 
     // send message
-    mavlink_msg_ekf_status_report_send(chan, flags, velVar, posVar, hgtVar, mag_max, temp, tasVar);
+    link.send_message(MAVLINK_MSG_ID_EKF_STATUS_REPORT, (const char*)&packet);
```

Comment:
can we use send_struct to keep type safety?

Reply:
Done